### PR TITLE
[REF] Refactor adding payment processor radio section onto register a…

### PR DESCRIPTION
**Pull request by seamuslee001**
…nd contribution main forms

Overview
----------------------------------------
Following the merge of https://github.com/civicrm/civicrm-core/pull/15940/files and picking up a comment in a DM from ages ago from @eileenmcnaughton i think we should try and refactor this a little

Before
----------------------------------------
Multiple places adding in the radio select option for Payment Processors

After
----------------------------------------
Single function

ping @mattwire

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I think the shared trait is a better place (avoids static functions too)

File: CRM/Core/Payment/Form.php [hunk 1]
```diff
@@ -412,4 +412,39 @@ public static function getPaymentLabel($processor) {
...
+    if (count($pamymentProcessors) > 1) {
+      $form->addRadio('payment_processor_id', ts('Payment Method'), $pamymentProcessors,
+        NULL, "&nbsp;", FALSE, $optAttributes
+      );
+    }
+    elseif (!empty($pamymentProcessors)) {
+      $ppKeys = array_keys($pamymentProcessors);
+      $currentPP = array_pop($ppKeys);
+      $form->addElement('hidden', 'payment_processor_id', $currentPP);
+      if ($currentPP === 0 && $formType === 'Contribute') {
```
Comment (eileenmcnaughton) [inline CRM/Core/Payment/Form.php]: Let's not do this - the Contribute form can do this part itself

Comment (seamuslee001) [reply CRM/Core/Payment/Form.php]: ok moved it back now @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
@seamuslee001 the trait I referred to is CRM_Financial_Form_FrontEndPaymentFormTrait

**Comment by seamuslee001** [comment]
@eileenmcnaughton at present on the event form is using the trait i think not contribution page

**Comment by seamuslee001** [comment]
actually take that back wrong

**Comment by eileenmcnaughton** [comment]
lol

**Comment by seamuslee001** [comment]
ok its on the trait now @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I worked through the same extraction & came up with a slightly different function

- the main difference is not passing the output of getProcessors as there is no cost to speak of in getting that from the form variables. Other than that the function name felt misleading, the abbreviated parameters can be made into meaningful ones 

```
  /**
   * Add the payment processor fields to the form.
   *
   *  Adds in either a set of radio buttons or hidden fields to contain the payment processors on a front end form
   */
  protected function addPaymentProcessorFieldsToForm() {
    $paymentProcessors = $this->getProcessors();
    $optAttributes = [];
    foreach (array_keys($paymentProcessors) as $key) {
      if ($key > 0) {
        $optAttributes[$key]['class'] = 'payment_processor_' . strtolower($this->_paymentProcessors[$key]['payment_processor_type']);
      }
      else {
        $optAttributes[$key]['class'] = 'payment_processor_paylater';
      }
    }
    if (count($paymentProcessors) > 1) {
      $this->addRadio('payment_processor_id', ts('Payment Method'), $paymentProcessors,
        NULL, '&nbsp;', FALSE, $optAttributes
      );
    }
    elseif (count($paymentProcessors) === 1) {
      $this->addElement('hidden', 'payment_processor_id', reset($paymentProcessors));
    }
  }
```

**Comment by seamuslee001** [comment]
@eileenmcnaughton just on the function name, I wanted to be clearer about the fact that it is the front end, on the back office fields it is a select field rather than radio button so as this function is only dealing with the front end adding of payment processors i wanted to be more explicit on that

backoffice-payment-processor

**Comment by eileenmcnaughton** [comment]
Doesn't the fact it's on the FrontEndPaymentFormTrait make that obsolete?

**Comment by seamuslee001** [comment]
I guess so

**Comment by seamuslee001** [comment]
have updated it now @eileenmcnaughton

File: CRM/Financial/Form/FrontEndPaymentFormTrait.php [hunk 2]
```diff
@@ -106,4 +106,30 @@ protected function getProcessors(): array {
...
+    foreach ($pamymentProcessors as $ppKey => $ppval) {
+      if ($ppKey > 0) {
+        $optAttributes[$ppKey]['class'] = 'payment_processor_' . strtolower($this->_paymentProcessors[$ppKey]['payment_processor_type']);
+      }
+      else {
+        $optAttributes[$ppKey]['class'] = 'payment_processor_paylater';
+      }
+    }
+    if (count($pamymentProcessors) > 1) {
+      $this->addRadio('payment_processor_id', ts('Payment Method'), $pamymentProcessors,
```
Comment (mattwire) [inline CRM/Financial/Form/FrontEndPaymentFormTrait.php]: @seamuslee001 Typo should be `$paymentProcessors`

Comment (seamuslee001) [reply CRM/Financial/Form/FrontEndPaymentFormTrait.php]: Thanks @mattwire fixed now

**Comment by seamuslee001** [comment]
@eileenmcnaughton is this good now?

**Pull request merged**